compiler/rustc_codegen_llvm/src/debuginfo/metadata.rs

```diff
@@ -1169,6 +1188,8 @@ fn build_closure_env_di_node<'ll, 'tcx>(
             Stub::Struct,
             unique_type_id,
             &type_name,
+            unknown_file_metadata(cx),
+            UNKNOWN_LINE_NUMBER,
```

Comment:
I think closures could have reasonable file info that would point to the line that contains the closure. Could you try `TyCtxt::def_span` here as well and see if that provides the correct info? (If it does, it's probably worth extracting a function that can be called from both locations)